scipy/optimize/optimize.py

```diff
@@ -126,6 +126,9 @@ def __repr__(self):
         else:
             return self.__class__.__name__ + "()"
 
+    def __dir__(self):
+        return self.keys()
+
```

Comment:
Looks like you need to return a sequence (not an iterable). https://docs.python.org/3/reference/datamodel.html#object.__dir__